app/jobs/repo_setup_job.rb

```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+require "concerns/repo_setup.rb"
+
+class RepoSetupJob < ApplicationJob
```

Comment:
If you want to handle the case of serialization failing (the assignment repo was deleted from the db before the job was performed), you can use `rescue_from ActiveJob::DeserializationError`